src/app/api/query-client.ts

```diff
@@ -0,0 +1,31 @@
+import { QueryClient } from "@tanstack/react-query";
+
+export const queryKeys = {
+  zones: {
+    list: ["zones"],
+  },
+} as const;
+
+type QueryKeyValue<T> = T extends (...args: any[]) => any ? ReturnType<T> : T;
+
+export type QueryKey = QueryKeyValue<
+  (typeof queryKeys)[keyof typeof queryKeys][keyof (typeof queryKeys)[keyof typeof queryKeys]]
```

Comment:
This was quite difficult to understand at first glance. If this is the cleanest way of doing it then I'm happy with that, but something more readable would be nice here if possible

Reply:
Fair point. Updated this.